jwst/datamodels/model_base.py

```diff
@@ -174,12 +174,11 @@ def __init__(self, init=None, schema=None, memmap=False,
                 else:
                     init_fitsopen = init
 
-                with fits.open(init_fitsopen, memmap=memmap) as hdulist:
-                    asdffile = fits_support.from_fits(hdulist,
-                                                  self._schema,
-                                                  self._ctx,
-                                                  **kwargs)
-                    self._files_to_close.append(hdulist)
```

Comment:
Perhaps the intent was that the underlying fits file wasn't meant to remain open after reading with `DataModel`? In that case this would make sense because you'd still need to be careful about closing the fits file even after using a context manager (especially with `memmap=True`) since `io.fits` makes it possible to access memory-mapped files even outside of the context. 

On the other hand, if the underlying fits file is supposed to remain accessible after reading in with a `DataModel`, then I agree, this doesn't make sense to me either.

Reply:
`hdulist` is not accessed anywhere else in __init__`.  The line is to extract the `AsdfFile` instance from the hdulist["ASDF"] extension and validate it using the given schema.  Once this is done, the hdulist is no longer needed here I think.

Would it make sense to leave the `with` context manager and not append `hdulist` to `self._files_to_close`?

Reply:
As per concensus during SCSB meeting [Datamodels, ASDF, and FITS efficiency](https://innerspace.stsci.edu/display/SCSB/Datamodels%2C+ASDF%2C+and+FITS+efficiency), the coded behavior will remain the same. Any resource opened by `DataModel` will remain open until `DataModel` is itself closed.